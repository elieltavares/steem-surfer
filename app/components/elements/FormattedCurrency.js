import React, { Component, Fragment } from 'react';

import PropTypes from 'prop-types';

class FormattedCurrency extends Component {
  render() {
    const { global, value, fixAt } = this.props;
    const { currencyRate, currencySymbol } = global;

    const valInCurrency = value * currencyRate;

    return (
      <Fragment key="result-val">{`${currencySymbol} ${valInCurrency.toFixed(
        fixAt
      )}`}</Fragment>
    );
  }
}

FormattedCurrency.defaultProps = {
  fixAt: 2
};

FormattedCurrency.propTypes = {
  global: PropTypes.shape({
    currencyRate: PropTypes.number.isRequired,
    currencySymbol: PropTypes.string.isRequired
  }).isRequired,
  value: PropTypes.number.isRequired,
  fixAt: PropTypes.number
};

export default FormattedCurrency;
