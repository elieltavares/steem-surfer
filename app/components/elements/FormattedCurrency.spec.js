/* eslint-disable */
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import FormattedCurrency from './FormattedCurrency';

Enzyme.configure({ adapter: new Adapter() });

describe('FormattedCurrency', () => {
  it('(1) render usd', () => {
    const props = {
      global: { currencyRate: 1, currencySymbol: '$' },
      value: 0.155
    };
    const component = shallow(<FormattedCurrency {...props} />);
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('(2) render euro', () => {
    const props = {
      global: { currencyRate: 0.84, currencySymbol: '€' },
      value: 0.155
    };
    const component = shallow(<FormattedCurrency {...props} />);
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('(3) render tl with 4 character after dot', () => {
    const props = {
      global: { currencyRate: 6.3, currencySymbol: '₺' },
      value: 0.155,
      fixAt: 4
    };
    const component = shallow(<FormattedCurrency {...props} />);
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
