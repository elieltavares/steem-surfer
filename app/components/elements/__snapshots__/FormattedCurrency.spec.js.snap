// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormattedCurrency (1) render usd 1`] = `"$ 0.15"`;

exports[`FormattedCurrency (2) render euro 1`] = `"€ 0.13"`;

exports[`FormattedCurrency (3) render tl with 4 character after dot 1`] = `"₺ 0.9765"`;
